tbot_contrib/gpio.py

```diff
@@ -0,0 +1,56 @@
+import tbot
+from tbot.machine import linux
+
+
+class Gpio:
+    def __init__(self, host: linux.LinuxShell, gpio_number: int):
+
+        self.host = host
+        self._gpio_number = str(gpio_number)
+        self._gpio_sysclass_path = self.host.fsroot / "sys/class/gpio"
+        self._gpio_path = self._gpio_sysclass_path / f"gpio{self._gpio_number}"
+        self._export()
+
+    def _export(self) -> None:
+        if not self._gpio_path.is_dir():
+            self.host.exec0(
+                "echo",
+                str(self._gpio_number),
+                linux.RedirStdout(self._gpio_sysclass_path / "export"),
+            )
+
+    def _is_in_direction(self) -> bool:
+        direction = self.host.exec0("cat", self._gpio_path / "direction")
+        return direction.strip().lower() == "in"
+
+    def set_in_direction(self) -> None:
+        if self._is_in_direction():
+            pass
+        else:
+            self.host.exec0(
+                "echo", "in", linux.RedirStdout(self._gpio_path / "direction")
+            )
+
+    def set_out_direction(self) -> None:
+        if not self._is_in_direction():
+            pass
+        else:
+            self.host.exec0(
+                "echo", "out", linux.RedirStdout(self._gpio_path / "direction")
+            )
```

Comment:
I'd create a single `set_direction()` method:

```suggestion
    def set_direction(self, direction: str) -> None:
        assert direction in ["in", "out"], f"Unsupported GPIO direction: {direction!r}"
        if self.direction == direction:
            return
            
        (self._gpio_path / "direction").write_text(direction)
        self.direction = direction
```